system-x/services/samba/src/main/java/software/tnb/samba/service/SambaServer.java

```diff
@@ -3,14 +3,18 @@
 import software.tnb.common.client.NoClient;
 import software.tnb.common.deployment.WithDockerImage;
 import software.tnb.common.service.Service;
-import software.tnb.common.validation.NoValidation;
 import software.tnb.samba.account.SambaAccount;
+import software.tnb.samba.validation.SambaValidation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.Map;
 
-public abstract class SambaServer extends Service<SambaAccount, NoClient, NoValidation> implements WithDockerImage {
+public abstract class SambaServer extends Service<SambaAccount, NoClient, SambaValidation> implements WithDockerImage {
```

Comment:
`NoClient` should be `SMBClient` (or perhaps `Session`?), so that you don't need to hassle with it in validation. Its instance should be created in `client()` method and the connection should be open/closed in `openResources` and `closeResources` methods (I don't know if that's possible or if there is a specific reason why you created a new connection/session every time in validation methods)